coordinator/src/orderbook/trading.rs

```diff
@@ -0,0 +1,191 @@
+        let order2 = dummy_order(dec!(21_000), 2, Default::default());
+        let order3 = dummy_order(dec!(20_500), 3, Default::default());
+
+        let orders = vec![order3.clone(), order1.clone(), order2.clone()];
+
+        let orders = sort_orders(orders, true);
+        assert_eq!(orders[0], order2);
+        assert_eq!(orders[1], order3);
+        assert_eq!(orders[2], order1);
+    }
+
+    #[test]
+    pub fn when_all_same_id_sort_by_id() {
+        let order1 = dummy_order(dec!(20_000), 1, Default::default());
+        let order2 = dummy_order(dec!(20_000), 2, Default::default());
+        let order3 = dummy_order(dec!(20_000), 3, Default::default());
+
+        let orders = vec![order3.clone(), order1.clone(), order2.clone()];
+
+        let orders = sort_orders(orders, true);
+        assert_eq!(orders[0], order1);
+        assert_eq!(orders[1], order2);
+        assert_eq!(orders[2], order3);
+
+        let orders = sort_orders(orders, false);
+        assert_eq!(orders[0], order1);
+        assert_eq!(orders[1], order2);
+        assert_eq!(orders[2], order3);
+    }
+
+    #[test]
+    fn given_limit_and_market_with_same_amount_then_match() {
+        let all_orders = vec![
+            dummy_order(dec!(20_000), 1, dec!(100)),
+            dummy_order(dec!(21_000), 2, dec!(200)),
+            dummy_order(dec!(20_000), 3, dec!(300)),
+            dummy_order(dec!(22_000), 4, dec!(400)),
+        ];
+
+        let order = Order {
+            id: 1,
+            price: Default::default(),
+            trader_id: "".to_string(),
+            taken: false,
+            direction: Direction::Short,
+            quantity: dec!(100),
+            order_type: OrderType::Market,
+        };
+
+        let matched_orders = match_order(order, all_orders).unwrap();
+
+        assert_eq!(matched_orders.len(), 1);
+        let matched_order = matched_orders.get(0).unwrap();
+        assert_eq!(matched_order.maker_order.quantity, dec!(100));
+    }
+
+    #[test]
+    fn given_limit_and_market_with_smaller_amount_then_match_multiple() {
+        let all_orders = vec![
+            dummy_order(dec!(20_000), 1, dec!(100)),
+            dummy_order(dec!(21_000), 2, dec!(200)),
+            dummy_order(dec!(22_000), 3, dec!(400)),
+            dummy_order(dec!(20_000), 4, dec!(300)),
+        ];
+
+        let order = Order {
+            id: 1,
+            price: Default::default(),
+            trader_id: "".to_string(),
+            taken: false,
+            direction: Direction::Short,
+            quantity: dec!(200),
+            order_type: OrderType::Market,
+        };
+
+        let matched_orders = match_order(order, all_orders).unwrap();
+
+        assert_eq!(matched_orders.len(), 2);
+        let matched_order = matched_orders.get(0).unwrap();
+        assert_eq!(matched_order.maker_order.id, 1);
+        let matched_order = matched_orders.get(1).unwrap();
+        assert_eq!(matched_order.maker_order.id, 4);
+    }
+}
```

Comment:
It would be cool to have a test that showcases that we match with opposite direction only.